src/libraries/System.Private.CoreLib/src/System/Collections/ObjectModel/ReadOnlyCollection.cs

```diff
@@ -229,4 +230,32 @@ void IList.RemoveAt(int index)
             ThrowHelper.ThrowNotSupportedException(ExceptionResource.NotSupported_ReadOnlyCollection);
         }
     }
+
+    /// <summary>
+    /// Provides static methods to create read-only collections and sets.
+    /// </summary>
+    public static class ReadOnlyCollection
+    {
+        /// <summary>
+        /// Creates a new <see cref="ReadOnlyCollection{T}"/> from the specified span of values.
+        /// This method (simplifies collection initialization)[/dotnet/csharp/language-reference/operators/collection-expressions]
+        /// to create a new <see cref="ReadOnlyCollection{T}"/> with the specified values.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the collection.</typeparam>
+        /// <param name="values">The span of values to include in the collection.</param>
+        /// <returns>A new <see cref="ReadOnlyCollection{T}"/> containing the specified values.</returns>
+        public static ReadOnlyCollection<T> CreateCollection<T>(params ReadOnlySpan<T> values) =>
+            values.IsEmpty ? ReadOnlyCollection<T>.Empty : new ReadOnlyCollection<T>(values.ToArray());
+
+        /// <summary>
+        /// Creates a new <see cref="ReadOnlySet{T}"/> from the specified span of values.
+        /// This method (simplifies collection initialization)[/dotnet/csharp/language-reference/operators/collection-expressions]
+        /// to create a new <see cref="ReadOnlySet{T}"/> with the specified values.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the collection.</typeparam>
+        /// <param name="values">The span of values to include in the collection.</param>
+        /// <returns>A new <see cref="ReadOnlySet{T}"/> containing the specified values.</returns>
+        public static ReadOnlySet<T> CreateSet<T>(params ReadOnlySpan<T> values) =>
```

Comment:
What is the reason for this method to be `ReadOnlyCollection.CreateSet<T>()` instead of `ReadOnlySet.Create<T>()` in a new static class?
The later is much more user friendly.

Reply:
The builder methods aren't for users. They are for compilers to build collection expressions.
The compiler only requires static method in non-generic type.

Reply:
> What is the reason for this method to be `ReadOnlyCollection.CreateSet<T>()` instead of `ReadOnlySet.Create<T>()` in a new static class? The later is much more user friendly.

I could replace `ReadOnlyCollection.CreateSet<T>()` with the `ReadOnlySet.Create<T>()` method if that's more correct.

Reply:
> I could replace `ReadOnlyCollection.CreateSet<T>()` with the `ReadOnlySet.Create<T>()` method if that's more correct.

That's not what's approved in API review. Questions around the method name should go back to the [original issue](https://github.com/dotnet/runtime/issues/110161).

Reply:
> > I could replace `ReadOnlyCollection.CreateSet<T>()` with the `ReadOnlySet.Create<T>()` method if that's more correct.
> 
> That's not what's approved in API review. Questions around the names should go back to the [original issue](https://github.com/dotnet/runtime/issues/110161).

It says that methods can be created as one class or in separate ones... https://github.com/dotnet/runtime/issues/110161#issuecomment-2515391504

> We could add another type (e.g. System.Collections.ObjectModel.Collection **or one non-generic type per collection** (like we do in immutable).